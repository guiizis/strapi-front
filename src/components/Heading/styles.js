import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
  `,
};

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)}
  `}
`;
