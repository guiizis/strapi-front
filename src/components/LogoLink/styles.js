import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-itens: center;
    text-deoration: none;
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
