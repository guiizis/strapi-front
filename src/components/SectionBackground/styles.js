import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;

const containerBackgroundActive = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;
