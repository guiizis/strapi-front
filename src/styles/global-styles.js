import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Montserrat:ital@1&family=Open+Sans:ital,wght@0,400;0,700;1,400;1,600&display=swap');

* {
    margin: 0;
    padding: 0;
    box-sizing: 'border-box'
  }

  html {
    font-size: 62.5%;
  }

  body {
    font-size: 2.4rem;
    font-family: 'Open Sans', sans-serif;
  }

  h1 {
    font-family: 'Montserrat', sans-serif;
  }
`;
