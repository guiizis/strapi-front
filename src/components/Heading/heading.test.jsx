import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('heading', () => {
  it('should render with default values', () => {
    renderTheme(<Heading> Texto </Heading>);

    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.xlarge,
      'text-transform': 'none',
    });
  });

  it('should render with white color', () => {
    renderTheme(<Heading colorDark={false}> Texto </Heading>);

    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render correct heading  sizes', () => {
    renderTheme(<Heading size="small"> Texto </Heading>);

    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
    });
  });

  it('should render correct font size when screen is mobile', () => {
    renderTheme(<Heading size="huge"> Texto </Heading>);

    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyleRule('font-size', theme.fonts.sizes.xlarge, {
      media: theme.media.lteMedium,
    });
  });

  it('should render with upperCase letters', () => {
    renderTheme(<Heading upperCase={true}> Texto </Heading>);

    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render a h2 tag', () => {
    renderTheme(<Heading as="h2"> Texto </Heading>);

    expect(
      screen.getByRole('heading', { level: 2, name: 'Texto' }),
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading> Texto </Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
