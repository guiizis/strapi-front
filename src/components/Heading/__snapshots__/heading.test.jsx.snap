// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading should match snapshot 1`] = `
.c0 {
  color: #0a1128;
  font-size: 4.0rem;
  text-transform: none;
}

<h1
  class="c0"
>
   Texto 
</h1>
`;
